redhat-appstudio/service-provider-integration-operator
# feat: Enhance error handling in the operator

### What does this PR do?
This enhances the error handling in the `SPIAccessToken` controller.

Introduce Invalid and Error phases, error reason and error message to the SPIAccessToken status.
    
Enhance the github metadata provider with the ability to distinguish between token errors and other "environmental" errors.

Additionally, this removes the `AuthenticatedOAuthState` from the codebase of the operator. It is only used by the oauth service so it has no place in the operator codebase. Instead, the code in the operator was made more flexible to support any kind of datastructure encoded as the OAuth state (which will be used by the OAuth service).
### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
This PR is a preparation for https://issues.redhat.com/browse/SVPI-62 - the endpoint for manual token upload

### How to test this PR?
1. Create an `SPIAccessToken` object with an invalid service provider URL.
2. The token phase should flip to "Invalid"

The other error states are currently not invokable using normal operation of the operator. This will only be possible once we have the manual upload endpoint.
